### ISSUE TITLE
Logitech hidpp reciever is getting stuck in bootloader mode

### ISSUE THREAD

**Issue by superm1** [fwupd/fwupd#1951]
**Describe the bug**

Running 'reinstall' on this device is causing it to timeout and get stuck in bootloader mode.
```
  DeviceId:             50c93db0cd70243d33c15121e7d67d644f1718e3
  Guid:                 9d131a0c-a606-580f-8eda-80587250b8d6
  Guid:                 fcf55bf5-767b-51ce-9c17-f6f538c4ee9f <- HIDRAW\VEN_046D&DEV_C52B&REV_00
  Guid:                 279ed287-3607-549e-bacc-f873bb9838c4 <- HIDRAW\VEN_046D&DEV_C52B
  Summary:              A miniaturised USB wireless receiver
  Plugin:               logitech_hidpp
  Protocol:             com.logitech.unifying
  Flags:                supported|registered|wait-for-replug
  VendorId:             USB:0x046D
  Version:              RQR12.10_B0032
  VersionBootloader:    BOT01.02_B0014
  VersionFormat:        plain
  Icon:                 preferences-desktop-keyboard
  InstallDuration:      30
  Created:              2020-04-03
  Modified:             2020-04-03
  UpdateState:          failed
  PhysicalId:            DEVPATH=/devices/pci0000:00/0000:00:1d.0/0000:03:00.0/0000:04:04.0/0000:3c:00.0/0000:3d:02.0/0000:3e:00.0/usb5/5-2/5-2.3/5-2.3.3/5-2.3.3:1.2
  SignedFirmware:        false
```

**Steps to Reproduce**
Steps to reproduce the behavior.

`fwupdtool reinstall --plugin-whitelist=logitech_hidpp $DEVICEID`

I did happen to get a segfault once as well, but didn't get it the second time.
[segfault.txt](https://github.com/fwupd/fwupd/files/4429209/segfault.txt)

However there appears to be some memory cleanup bug problem with the connected devices (which is probably new due to `EnumerateAllDevices=true`.

```
18:22:05:0732 FuDeviceList         doing delayed removal
18:22:05:0732 FuDeviceList         ::removed 224b61aa4fea82347a28e1bb02c77ca689e38987
18:22:05:0732 FuMain               REMOVED:
FuLogitechHidPpPeripheral:
Logitech M510
  DeviceId:             224b61aa4fea82347a28e1bb02c77ca689e38987
  ParentDeviceId:       d14ced0fa19e6da7ca24660d2a590854f1ebb8c9
  Guid:                 47ca0e4f-54da-52bf-91dd-037713db1600 <- HIDRAW\VEN_046D&DEV_4051&REV_00
  Guid:                 ff0cf62d-c010-5051-876a-1203132289e8 <- HIDRAW\VEN_046D&DEV_4051
  Guid:                 0a44c29c-c138-59ad-ade1-cbe0c8b64ce9 <- UFY\VID_046D&PID_4051
  Summary:              Unifying Mouse
  Plugin:               logitech_hidpp
  Protocol:             com.logitech.unifying
  Flags:                registered|only-supported
  VendorId:             USB:0x046D
  Version:              RQM62.01_B0015
  VersionFormat:        plain
  Icon:                 input-mouse
  Created:              2020-04-03
  PhysicalId:            HID_PHYS=usb-0000:3e:00.0-2.3.3/input2:2
  HidppVersion:          4
  HidppId:               0x2
  BatteryLevel:          70
  IsUpdatable:           false
  IsActive:              true
  Feature00:             Root [0x0000]
  Feature03:             GetDevicenameType [0x0005]
  Feature02:             IFirmwareInfo [0x0003]
  Feature05:             BatteryLevelStatus [0x1000]

18:22:05:0734 GLib-GObject         invalid unclassed pointer in cast to 'FwupdDevice'
18:22:05:0734 Fwupd                fwupd_device_has_flag: assertion 'FWUPD_IS_DEVICE (device)' failed
```

**Comment by hughsie** [comment]
> fwupdtool reinstall

Also with `fwupdmgr`?

> 18:22:05:0734 GLib-GObject         invalid unclassed pointer in cast to 'FwupdDevice'

Can you set `G_DEBUG=fatal-criticals` and then run it under gdb, and get a backtrace of where this happens? TIA.

**Comment by hughsie** [comment]
> segfault

    18:16:29:0510 FuPluginLogitechHidPp HID++ ID is fe
    18:16:29:0511 FuPluginLogitechHidPp added feature GetDevicenameType [0x0005] as idx 5a

Doesn't seem plausible...

    firmware entity count is 255

That's wrong..

    firmware entity 0x00 version is ���ff.ff_Bffff

...maybe the output from valgrind too? We probably need to change the blacklist of g_ascii_isspace() to something like g_ascii_isprint() in fu_logitech_hidpp_format_version although that's not addressing the root issue.

**Comment by superm1** [comment]
I've discovered that I can only reproduce this when two receivers are connected (the second doesn't have any paired devices though).  Some of the weird debugging (large entities and garbage) don't show up with just one receiver and it properly flashes and recovers.

I still haven't gotten the segfault again, but I did run through valgrind (see attached).
The command I ran was
`sudo valgrind --leak-check=full fwupdtool --plugin-whitelist=logitech_hidpp reinstall 50c93db0cd70243d33c15121e7d67d644f1718e3 --verbose 2>&1| tee ~/fwupd.txt`
[fwupd.txt](https://github.com/fwupd/fwupd/files/4432919/fwupd.txt)
